catroid/src/main/java/org/catrobat/catroid/ui/ProjectUploadActivity.java

```diff
@@ -263,6 +278,62 @@ private void onNextButtonClick() {
 		}
 	}
 
+	private void checkCodeforApiKey() {
+		String regex = "(<value>).*?((?=.*[A-Za-z].*[0-9].*)[A-Za-z0-9]{24,45}).*?(</value>)";
+		xmlFile = new File(project.getDirectory(), CODE_XML_FILE_NAME);
+
+		try {
+			xml = com.google.common.io.Files.toString(xmlFile, Charsets.UTF_8);
+		} catch (IOException e) {
+			Log.e(TAG, Log.getStackTraceString(e));
+		}
+		if (xml.contains(WEB_REQUEST_BRICK) || xml.contains(BACKGROUND_REQUEST_BRICK) || xml.contains(LOOK_REQUEST_BRICK) || xml.contains(OPEN_URL_BRICK)) {
+			apiPattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+			apiMatcher = apiPattern.matcher(xml);
+			if (apiMatcher.find()) {
+				showApiReplacementDialog(Objects.requireNonNull(apiMatcher.group(2)));
+			}
+		}
+	}
+
+	private void apiKeyFound() {
+		if (apiMatcher.find(apiMatcher.end())) {
+			showApiReplacementDialog(Objects.requireNonNull(apiMatcher.group(2)));
+		}
+	}
+
+	private void replaceSecret(String secrete) {
+		xml = xml.replaceAll(secrete, "INSERT-API-KEY-HERE");
```

Comment:
While reviewing, Julia and I talked about the translatable part, is it really convenient to translate that part? If the key is being replaced by a certain language(say Russian) and the person that downloads the project does not speak that specific language, they wouldn´t know what happened there and do not know what/that it should be replaced.

Reply:
> While reviewing, Julia and I talked about the translatable part, is it really convenient to translate that part? If the key is being replaced by a certain language(say Russian) and the person that downloads the project does not speak that specific language, they wouldn´t know what happened there and do not know what/that it should be replaced. While the English version would be understandable for most languages

Three reasons:
1. Users *primarily* code for themselves. 
2. Russian projects will anyway contain a lot of Russian, e.g., variable names, name of project, strings in parameters, etc.
3. We are already thinking of ways to include an optional in-app translate option allowing the user to see translations for all texts that might be in another language. This will include texts in strings in parameters. 

By the way, to make it easier for our own translators, but also to facilitate point 3 above: Please change the inserted text to "insert API key here", i.e., without the all-caps and with dashes in between.

Reply:
Ok that makes sense. I changed the inserted string to "insert API key here"!